give some advice how to associate VN and Retry with connections (#213)
quicwg/qlog: draft-ietf-quic-qlog-quic-events.md

Ideally, implementers SHOULD create a separate, individual "endpoint-level" trace
file (or group_id value), not associated with a specific connection (for example a
"server.qlog" or group_id = "client"), and log all events that do not belong to a
single connection to this grouping trace.

There are events that are conceptually associated with a single connection, but
that happen before the actual QUIC connection is established when viewed from
the server's perspective. This includes version negotiation as well as retries.

> Servers MAY take extra efforts to associate these events with a successfully
> established QUIC connection and combine them in a single trace. For version
> negotiation, matching can be done by identifying a connection attempt by its
> four-tuple, and retries can be identified by their ODCID.

Reviewer: I don't quite  like the way the text read says "MAY take extra effort". Mainly because this text is also there to be read by consumers who want to know what events to expect but don't care about effort. 

A suggested rewording to avoid. Other options would work just too.

```suggestion
Servers MAY include such events in the connection trace but need to ensure those
events are correctly associated with the correct connection, which can take
extra effort or resource. For example, version negotiation events can be matched
to a connection by identifying a connection attempt by its four-tuple;
similarly, retry events be matched by their ODCID.
```

Author: Suggestion sounds good. Thank you!

Ideally, implementers SHOULD create a separate, individual "endpoint-level" trace
file (or group_id value), not associated with a specific connection (for example a
"server.qlog" or group_id = "client"), and log all events that do not belong to a
single connection to this grouping trace.

There are events that are conceptually associated with a single connection, but
that happen before the actual QUIC connection is established when viewed from
the server's perspective. This includes version negotiation as well as retries.
Servers MAY include such events in the connection trace but need to ensure those
events are correctly associated with the correct connection, which can take

> extra effort or resource. For example, version negotiation events can be matched

Author: ```suggestion
extra effort or resources. For example, version negotiation events can be matched
```

to a connection by identifying a connection attempt by its four-tuple;

> similarly, retry events be matched by their ODCID.

Reviewer: think this was my fault, sorry

```suggestion
similarly, retry events can be matched by their ODCID.
```

> 

Reviewer 2: In a server farm, there is no guarantee that VN and later connection are handled by the same server -- in contrast with the ODCID mechanism that can be used to achieve that. That means it could be hard to associate VN and connection at logging time. And then, the is also no guarantee that either Retry or VN is followed by an actual connection. Retry may be caused by a DDOS attack. VN can be caused by something like [quicping](https://github.com/ooni/probe/issues/1994). In fact, I could see a need for an easy way to track orphans, and see how the world is using or abusing the server. Something like a global log, with events like VN, Retry, Connection arrival, Connection final.

Reviewer: This makes sense. I think it is already covered by the preceeding paragraph though? Maybe what we need to say is that events can be put in multiple logs - the global one, and the conenction-specific one if the match succeeds.
